mail/views.py

```diff
@@ -237,3 +239,64 @@ def post(self, request, *args, **kwargs):
             status=mailgun_response.status_code,
             data=generate_mailgun_response_json(mailgun_response)
         )
+
+
+class EmailBouncedView(APIView):
+    """
+    View class that handles HTTP requests to capture bounced email
+    """
+    permission_classes = (MailGunWebHookPermission, )
+
+    @classmethod
+    def fetch_custom_var(cls, headers, keyword):
+        """
+        fetch data from message-headers
+
+        Args:
+            headers (Json): complete Json object
+            keyword (str): json key that need to fetch
+
+        Returns:
+            Json: return json object if matches
+        """
+        var_dict = None
+        for header in headers:
+            if header[0] == keyword:
+                var_dict = header[1]
+
+        return var_dict
+
+    def post(self, request, *args, **kargs):  # pylint: disable=unused-argument
+        """
+        POST method handler
+        """
+        event = request.POST.get("event", None)
+        recipient = request.POST.get("recipient", None)
+        error = request.POST.get("error", None)
+        message_headers = request.POST.get("message-headers", None)
+        error_msg = (
+            "Email to: {to} is bounced with an error message {error}".format(
+                to=recipient,
+                error=error
+            )
+        )
+
+        # load custom vars from headers
+        headers_obj = json.loads(message_headers)
+        custom_vars = EmailBouncedView.fetch_custom_var(headers_obj, "X-Mailgun-Variables")
```

Comment:
As you mentioned earlier the variables will show up in the post body, so it seems like we should just look there for the `log_error_on_bounce` value.